### PR TITLE
Reports App fixes and refactoring

### PR DESCRIPTION
**Pull request by dhavalarora**
**# Why?**
Download report should be in Reports app and shouldn't be there on Dashboard app.

**# Changes**
- Refactoring of existing codebase of Reports app
- Moved Download Report API from Dashboard App to Reports App.

### CODE REVIEW

File: apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_controller.ex [hunk 1]
```diff
@@ -322,29 +322,6 @@ defmodule AcqdatApiWeb.DashboardManagement.DashboardController do
     end
   end
 
-  def reports(conn, params) do
```
Comment (BandanaPandey) [inline apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_controller.ex:325]: @dhavalarora  why did you remove reports action from here, this particular feature is getting used on dashboard to download the reports ?
It is different than our reports app

File: apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_export_controller.ex [hunk 2]
```diff
@@ -200,33 +200,6 @@ defmodule AcqdatApiWeb.DashboardManagement.DashboardExportController do
     end
   end
 
-  def reports(conn, params) do
```
Comment (BandanaPandey) [inline apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_export_controller.ex:203]: here also @dhavalarora  why did you remove reports action from here, this particular feature is getting used to download the reports ?

File: apps/acqdat_api/lib/acqdat_api_web/controllers/reports/template_controller.ex [hunk 3]
```diff
@@ -4,29 +4,41 @@ defmodule AcqdatApiWeb.Reports.TemplateController do
...
     changeset = verify_index_params(params)
-    {:ok, data} = extract_changeset_data(changeset)
-    templates = Templates.get_all(data)
 
-    conn
-    |> put_status(200)
-    |> render("index.json", %{templates: templates})
+    with nil <- conn.status,
+         {:ok, data} <- extract_changeset_data(changeset),
+         templates <- TemplateModel.get_all(data) do
```
Comment (BandanaPandey) [inline apps/acqdat_api/lib/acqdat_api_web/controllers/reports/template_controller.ex:19]: @dhavalarora  we don't directly use model functions here in controller, we use it through the context AcqdatApi.Reports.Template specified here (as def delegate).
Please make sure to no use model functions directly here, we follow this convention throughout our applications

File: apps/acqdat_api/lib/acqdat_api_web/controllers/reports/template_instance_controller.ex [hunk 4]
```diff
@@ -6,32 +6,38 @@ defmodule AcqdatApiWeb.Reports.TemplateInstanceController do
...
     changeset = verify_index_params(params)
-    {:ok, data} = extract_changeset_data(changeset)
-    template_instances = TemplateInstance.get_all(data)
 
-    conn
-    |> put_status(200)
-    |> render("index.json", %{template_instances: template_instances})
+    with nil <- conn.status,
+         {:ok, data} <- extract_changeset_data(changeset),
+         template_instances <- TemplateInstanceModel.get_all(data) do
```
Comment (BandanaPandey) [inline apps/acqdat_api/lib/acqdat_api_web/controllers/reports/template_instance_controller.ex:18]: Please don't call model functions directly, define in this file AcqdatApi.Reports.TemplateInstance
and then use it through it

**Review by BandanaPandey — CHANGES_REQUESTED** [review]

File: apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_controller.ex (hunk 1 above)
Comment (chandradot99) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_controller.ex:325]: @BandanaPandey we are moving download report feature from dashboard app and moving it to reports app.

File: apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_export_controller.ex (hunk 2 above)
Comment (chandradot99) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_export_controller.ex:203]: @BandanaPandey same here also, we don't need this.

File: apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_controller.ex (hunk 1 above)
Comment (BandanaPandey) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_controller.ex:325]: ok @chandradot99

File: apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_export_controller.ex (hunk 2 above)
Comment (BandanaPandey) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/dashboard_management/dashboard_export_controller.ex:203]: ok @chandradot99

**Review by BandanaPandey — APPROVED** [review]

**Pull request merged**